src/caches.jl

```diff
@@ -131,20 +140,37 @@ function OrdinaryDiffEq.alg_cache(
 
     # Measurement model related things
     R = zeros(uElType, d, d)
-    H = zeros(uElType, d, D)
+    H = if KRONECKER
+        copy(E1)
```

Comment:
Check if it makes a difference if H is already some correct matrix (E1) or just initialized empty (zeros). This fact should really not depend just on KRONECKER